aeron-archive/src/main/resources/archive/aeron-archive-codecs.xml

```diff
@@ -572,6 +572,14 @@
         <field name="recordingId"          id="3" type="int64"/>
     </sbe:message>
 
+    <sbe:message name="ArchiveIdRequest"
```

Comment:
I was wondering if instead of (or as well as) having a separate request to query the archiveId, could we update the archive response message to always carry the archive id with it.  Then it would come back to the client during the initial connect stage and could be cached on the archive client.  It would also mean that it is always available in response to requests done via the proxy, e.g. startRecording so that it can be used immediately and not need an additional round trip.

Reply:
That's an interesting point. This will have to be done in  backwards compatible way. Perhaps adding `ControlResponse2` could be an option.

Reply:
Would we need a `ControlResponse2` to support older clients connecting to newer Archives?  Is that something we expect to support?

Reply:
@mikeb01 I think supporting old clients is important. I tried adding an optional field to the `ControlResponse` but that will not work for old clients connecting to the Archive.
```xml
<field name="archiveId" id="7" type="int64" presence="optional" sinceVersion="9" description="Id of the Archive sending the response."/>
```
Given the complexities of extending the response I suggest we keep a separate command instead.